### ISSUE TITLE
Cannot use with cuda in Window

### ISSUE THREAD

**Issue by thunanguyen** [mratsim/Arraymancer#409]
Hi, I cannot compile wwith the flag -d:cudnn in Window

My computer use:
GTX 1060
cuda 10.0
cudnn 7

**Comment by mratsim** [comment]
Yes I didn't configure Cuda/Cudnn for windows. I didn't have a Windows PC with a Nvidia GPU when I wrote the Cuda backend.

Hopefuly the only thing needed is to change the paths to the compiler here:

https://github.com/mratsim/Arraymancer/blob/5bf4327bd3b82cbb498f236abfe9e815c7616059/nim.cfg#L3-L32

**Comment by thunanguyen** [comment]
thanks, I've found that and changed already. But I wonder is there any ways to a unified config file for all OS ?

**Comment by mratsim** [comment]
Ideally I want to completely do away with the config files. I don't like them at all but at the moment they are a necessary evil.

Let's go over the config file:

https://github.com/mratsim/Arraymancer/blob/71ccad01953060fad121a03b49b97e81ed8ac12c/nim.cfg#L3-L32

This could be removed if NVCC becomes an official compiler in Nim with the proper flags.
This would be configured here I guess: https://github.com/nim-lang/Nim/blob/devel/config/nim.cfg

There is still one issue left though: either I distribute precompiled CUDA-enabled binaries or users ensure that CUDA/nvcc is in their path which AFAIK isn't automatic on Windows.

I tried using Clang but Clang support always has a versioning delay with CUDA, see https://github.com/mratsim/Arraymancer/pull/372#issue-310688322 when I tried in August

| clang++ | supported CUDA release | supported SMs |
| ------- | ---------------------- | ------------- |
| 3.9-5.0 | 7.0-8.0                | 2.0-(5.0)6.0  |
| 6.0     | [7.0-9.0](https://github.com/llvm-mirror/clang/blob/release_60/include/clang/Basic/Cuda.h) | [(2.0)3.0-7.0](https://github.com/llvm-mirror/clang/blob/release_60/lib/Basic/Targets/NVPTX.cpp#L163-L188) |
| 7.0     | [7.0-9.2](https://github.com/llvm-mirror/clang/blob/release_70/include/clang/Basic/Cuda.h) | [(2.0)3.0-7.2](https://github.com/llvm-mirror/clang/blob/release_70/lib/Basic/Targets/NVPTX.cpp#L196-L223) |
| 8.0     | [7.0-10.0](https://github.com/llvm-mirror/clang/blob/release_80/include/clang/Basic/Cuda.h) | [(2.0)3.0-7.5](https://github.com/llvm-mirror/clang/blob/release_70/lib/Basic/Targets/NVPTX.cpp#L199-L228) |
| trunk   | [7.0-10.1](https://github.com/llvm-mirror/clang/blob/master/include/clang/Basic/Cuda.h) | [(2.0)3.0-7.5](https://github.com/llvm-mirror/clang/blob/master/lib/Basic/Targets/NVPTX.cpp#L200-L229) |

Apparently the new Clang 9.1 supports Cuda 10.1 (though Clang is still annoying to install on Windows)

https://github.com/mratsim/Arraymancer/blob/71ccad01953060fad121a03b49b97e81ed8ac12c/nim.cfg#L34-L41

and

https://github.com/mratsim/Arraymancer/blob/71ccad01953060fad121a03b49b97e81ed8ac12c/nim.cfg#L47-L57

Would require to replace BLAS and LAPACK which is something I've started to do in [Laser](https://github.com/numforge/laser) but besides the matrix multiplication which is now as fast as OpenBLAS in pure Nim, I need QR decomposition, LU decomposition, Eigenvalues and solver implemented as well (from https://github.com/mratsim/Arraymancer/tree/master/src/linear_algebra/helpers)

https://github.com/mratsim/Arraymancer/blob/71ccad01953060fad121a03b49b97e81ed8ac12c/nim.cfg#L59-L67

would require removing OpenMP, for which [Weave](https://github.com/mratsim/weave) is a very viable alternative and actually probably much better as it does not suffer from: https://github.com/zy97140/omp-benchmark-for-pytorch

https://github.com/mratsim/Arraymancer/blob/71ccad01953060fad121a03b49b97e81ed8ac12c/nim.cfg#L69-L81

This isn't necessary anymore since: https://github.com/nim-lang/Nim/issues/12662
I did [implement the changes in Weave's benchmarks](https://github.com/mratsim/weave/blob/3bfb6416ba312bdcd066704fa897ddfa5fae1e6f/benchmarks/matmul_gemm_blas/gemm_pure_nim/common/gemm_ukernel_avx512.nim#L6) and Arraymancer can use a similar scheme in the same files.
